plugins/chat/app/controllers/chat_controller.rb

```diff
@@ -98,20 +98,22 @@ def create_message
 
     content = params[:message]
 
-    chat_message_creator =
-      Chat::ChatMessageCreator.create(
-        chat_channel: @chat_channel,
-        user: current_user,
-        in_reply_to_id: reply_to_msg_id,
-        content: content,
-        staged_id: params[:staged_id],
-        upload_ids: params[:upload_ids],
-      )
+    create_payload = Chat::ChatMessageCreatorV2::ChatMessageCreatorPayload.new.tap do |payload|
+      payload.in_reply_to_id = reply_to_msg_id
+      payload.content = content
+      payload.staged_id = params[:staged_id]
+      payload.upload_ids = params[:upload_ids]
+    end
+    create_result = Chat::ChatMessageCreatorV2.new(guardian, @chat_channel, create_payload).execute
 
-    return render_json_error(chat_message_creator.error) if chat_message_creator.failed?
+    if create_result.failed?
+      return render_json_error(create_result.errors), status: create_result.http_status
+    end
 
+    # NOTE: All of this should probably be in the service too?? It's all to do with
```

Comment:
You will need to look at the controller to see, but we are doing heaps of stuff here _after_ creating the message. Maybe that should be a separate service e.g. `NewChatMessageProcessor` or something like that...not sure